### ISSUE TITLE
Logical issue with `{}` type

### ISSUE THREAD

**Issue by avin-kavish** [typescript-eslint/typescript-eslint#5542]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Not sure which category to fit this in.

When I use `{}` as a type, it tells me 'don't use `{}` as a type, it means non-nullish value'. Except, I use it because I'm exactly expecting a non-nullish value, and this is the type that means non-nullish. Typescript 4.8 is also going to be switching to this to indicate non-nullability, so I think this message or rule needs to be reconsidered.

### Affected URL(s)

None.

**Comment by bradzacher** [comment]
Thanks for the issue. You are one of the few people that use it correctly.

Most people fundamentally misunderstand TypeScript's object model and assume that `{}` means "an object with no properties", because that's what it means when the same syntax is used as a value.

Which is why you'll see a lot of people write code like this:

```ts
function Foo(args: {}) {
  // ... 
}
```

Not know this function accepts any object, string, number, etc.

Thus the rule seeks to educate by flagging this.

There is an example in the docs of how you can disable just that part of the default config:
https://typescript-eslint.io/rules/ban-types/#extenddefaults
